controllers/informationCenter/SubmissionInformationCenterHandler.inc.php

```diff
@@ -19,6 +19,43 @@
 
 class SubmissionInformationCenterHandler extends InformationCenterHandler {
 
+	/** @var boolean Is the current user assigned to an editorial role for this submission */
+	var $_isCurrentUserAssignedEditor;
+
+	/**
+	 * @copydoc PKPHandler::authorize()
+	 */
+	function authorize($request, &$args, $roleAssignments) {
+		$success = parent::authorize($request, $args, $roleAssignments);
+
+		// Prevent users from accessing history unless they are assigned to an
+		// appropriate role in this submission
+		$this->_isCurrentUserAssignedEditor = false;
+		$userAssignedRoles = $this->getAuthorizedContextObject(ASSOC_TYPE_ACCESSIBLE_WORKFLOW_STAGES);
+		if (!empty($userAssignedRoles)) {
+			foreach ($userAssignedRoles as $stageId => $roles) {
+				if (array_intersect(array(ROLE_ID_SITE_ADMIN, ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR), $roles)) {
+					$this->_isCurrentUserAssignedEditor = true;
+					break;
+				}
+			}
+		} else {
+			$userGlobalRoles = $this->getAuthorizedContextObject(ASSOC_TYPE_USER_ROLES);
+			if (array_intersect(array(ROLE_ID_SITE_ADMIN, ROLE_ID_MANAGER), $userGlobalRoles)) {
+				$this->_isCurrentUserAssignedEditor = true;
+			}
+		}
+
+		if (!$this->_isCurrentUserAssignedEditor) {
```

Comment:
hmmm... somehow I do not understand what we are doing here: 
ROLE_ID_SITE_ADMIN and ROLE_ID_MANAGER can access it, if assigned or not.
Thus it seems this only looks if the ROLE_ID_SUB_EDITOR is assigned and only then it allows it to access the viewHistory.
But, I believe that all that is already cleared in the parent class, or? -- The parent::authorize allows only ROLE_ID_SUB_EDITOR, ROLE_ID_MANAGER and there is also the SubmissionAccessPolicy in the parent class (that checks the if the sub editor is assigned). Or? Hmmm...

Reply:
Hmm... looking back at this, I'm not sure. But I _think_ that this catches a situation where someone might have `ROLE_ID_MANAGER` in the context but is assigned as `ROLE_ID_AUTHOR` to the submission.

Looking at `SubmissionAccessPolicy`, it does call `UserAccessibleWorkflowStageRequiredPolicy` for sub editors and authors, which should deny access to the `viewHistory` op. But for managers, they only get `RoleBasedHandlerOperationPolicy`, which I don't think checks against submission assignments.

So, my thinking is that a manager who is only assigned as an author could get passed the authorization checks to this stage. If so, this code is necessary as a last check.

I'm not sure though. It may have just been a defensive addition because I wasn't sure.

Reply:
yes, ok :+1:

Reply:
hmmm... why did we then anonymize event logs when such an author will however not be able to access this then, or? hmmm...

Reply:
Ah, sorry, I see -- if a user is assigned as editor and author, it will be able to see the history, if it is only assigned as author (but has generally also editor role) here it is prevented for him to see the history
OK, everything clear...

Reply:
:+1: